src/types/multiroot.rs

```diff
@@ -0,0 +1,294 @@
+    /// precision (\epsilon \approx 2.22 \times 10^{-16}). The order of convergence of Newton’s
+    /// algorithm is quadratic, but the finite differences require n^2 function evaluations on
+    /// each iteration. The algorithm may become unstable if the finite differences are not a
+    /// good approximation to the true derivatives.
+    pub fn dnewton() -> MultiRootFSolverType {
+        ffi_wrap!(gsl_multiroot_fsolver_dnewton)
+    }
+
+    /// The Broyden algorithm is a version of the discrete Newton algorithm which attempts to
+    /// avoids the expensive update of the Jacobian matrix on each iteration. The changes to
+    /// the Jacobian are also approximated, using a rank-1 update,
+    ///
+    ///```text
+    ///J^{-1} \to J^{-1} - (J^{-1} df - dx) dx^T J^{-1} / dx^T J^{-1} df
+    ///```
+    ///
+    /// where the vectors dx and df are the changes in x and f. On the first iteration the inverse
+    /// Jacobian is estimated using finite differences, as in the discrete Newton algorithm.
+    ///
+    /// This approximation gives a fast update but is unreliable if the changes are not small, and
+    /// the estimate of the inverse Jacobian becomes worse as time passes. The algorithm has a
+    /// tendency to become unstable unless it starts close to the root. The Jacobian is refreshed
+    /// if this instability is detected (consult the source for details).
+    ///
+    /// This algorithm is included only for demonstration purposes, and is not recommended for
+    /// serious use.
+    pub fn broyden() -> MultiRootFSolverType {
+        ffi_wrap!(gsl_multiroot_fsolver_broyden)
+    }
+}
+
+ffi_wrapper!(
+    MultiRootFSolver,
+    *mut sys::gsl_multiroot_fsolver,
+    gsl_multiroot_fsolver_free
+    ;inner_call: sys::gsl_multiroot_function_struct => sys::gsl_multiroot_function_struct{ f: None, n: 0, params: std::ptr::null_mut() };,
+    "This is a workspace for multidimensional root-finding without derivatives."
+);
+
+impl MultiRootFSolver {
+    /// This function returns a pointer to a newly allocated instance of a solver of type `T` with
+    /// `n` unknowns.
+    ///
+    /// If there is insufficient memory to create the solver then the function returns a null
+    /// pointer and the error handler is invoked with an error code of `Value::NoMemory`.
+    #[doc(alias = "gsl_multiroot_fsolver_alloc")]
+    pub fn new(t: &MultiRootFSolverType, n: usize) -> Option<MultiRootFSolver> {
+        let ptr = unsafe { sys::gsl_multiroot_fsolver_alloc(t.unwrap_shared(), n) };
+
+        if ptr.is_null() {
+            None
+        } else {
+            Some(MultiRootFSolver::wrap(ptr))
+        }
+    }
+
+    /// This function initializes, or reinitializes, an existing solver `s` to use the multi
+    /// function `f` with `n` unknowns.
+    #[doc(alias = "gsl_multiroot_fsolver_set")]
+    pub fn set<F: Fn(&::VectorF64, &mut ::VectorF64) -> ::Value>(
+        &mut self,
+        f: F,
```

Comment:
You also need to store this closure otherwise it might be dropped before being used.